ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdapter.java

```diff
@@ -136,27 +128,34 @@ void requestBannerAd(@NonNull AdSize adSize, @NonNull MediationAdRequest mediati
     if (TextUtils.isEmpty(mPlacementId)) {
       String message = "Failed to load ad from Vungle: Missing or Invalid Placement ID.";
       Log.w(TAG, message);
-      if (bannerListener != null) {
-        bannerListener.onAdFailedToLoad(adapter, AdRequest.ERROR_CODE_INVALID_REQUEST);
-      }
-      return;
+      mVungleBannerListener.onAdFailedToLoad(AdRequest.ERROR_CODE_INVALID_REQUEST);
+      return null;
     }
 
     AdConfig.AdSize vungleAdSize = getSupportedAdSize(context, adSize);
     if (vungleAdSize == null) {
       String message = "Failed to load ad from Vungle: Invalid banner size.";
       Log.w(TAG, message);
-      if (bannerListener != null) {
-        bannerListener.onAdFailedToLoad(adapter, AdRequest.ERROR_CODE_INVALID_REQUEST);
-      }
-      return;
+      mVungleBannerListener.onAdFailedToLoad(AdRequest.ERROR_CODE_INVALID_REQUEST);
+      return null;
     }
+
+    mUniquePubRequestId = config.getRequestUniqueId();
+    if (VungleManager.getInstance().isBannerAdActive(mPlacementId, mUniquePubRequestId)) {
+      // Adapter does not support multiple Banner instances playing for same placement except for
+      // Refresh
+      mVungleBannerListener.onAdFailedToLoad(AdRequest.ERROR_CODE_INVALID_REQUEST);
+      return null;
+    }
+
     mAdConfig = VungleExtrasBuilder.adConfigWithNetworkExtras(mediationExtras, true);
     mAdConfig.setAdSize(vungleAdSize);
+    Log.d(TAG, "Requesting banner with ad size: " + mAdConfig.getAdSize());
+    mPendingRequestBanner = true;
```

Comment:
does mPendingRequestBanner need to be AtomicBoolean as we are checking for it in multiple places as its possible to get multiple AdRequests at same time

Reply:
all the ad request using the same thread, no need to dealers with AtomicBoolean